swap-frontend/src/SwapForm.tsx

```diff
@@ -185,29 +195,57 @@ export const SwapForm: Component = () => {
         }
         try {
             if (swapType() === 'in') {
-                const swap = await swapInService.createSwap(form.lightningInvoice);
+                const swap = await swapInService.createSwap(form.payload);
                 navigate(`/swap/in/${swap.swapId}`);
             } else if (swapType() === 'out') {
-                const swap = await swapOutService.createSwap(form.bitcoinAddress, inputAmount());
+                const swap = await swapOutService.createSwap(form.payload, inputAmount());
                 navigate(`/swap/out/${swap.swapId}`);
             }
         } catch (e) {
             toast.error('Unknown error');
         }
     }
 
+    createEffect(() => {
+        if (validated()) {
+            validate();
+        }
+    });
+
+    createEffect(() => {
+        const toAsset = form.to;
+        if (toAsset === 'ON_CHAIN_BITCOIN' || toAsset === 'ON_CHAIN_LIQUID') {
+            setSwapType('out');
+        } else if (toAsset === 'LIGHTNING_BITCOIN') {
+            setSwapType('in');
+        }
+    }, [form.from, form.to]);
+
```

Comment:
I think `swapType` doesn't need to be a signal anymore, does it?
It could just be a function:
```suggestion
        const toAsset = form.to;
        if (toAsset === 'ON_CHAIN_BITCOIN' || toAsset === 'ON_CHAIN_LIQUID') {
             return 'out';
        } else if (toAsset === 'LIGHTNING_BITCOIN') {
            return 'in';
        }
```

Reply:
Done